packages/opentelemetry-types/src/trace/status.ts

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * The status of a Span by providing a standard CanonicalCode in conjunction
+ * with an optional descriptive message.
+ */
+export interface Status {
+  /** The canonical code of this message. */
+  code: CanonicalCode;
+  /** A developer-facing error message. */
+  message?: string;
+}
+
+/** An enumeration of canonical status codes. */
+export enum CanonicalCode {
```

Comment:
Does it make sense to have a fixed enum for this? What if the reason falls outside of the range of values?

Also, should there be a code for exceptions, with the error information attached?

Reply:
I think will revise after https://github.com/open-telemetry/opentelemetry-specification/issues/59. Currently it is copied from OpenCensus project.